### PR TITLE
Init PR

### PR DESCRIPTION
**Pull request by ondrajodas**
Solving - https://keboola.atlassian.net/browse/COM-73

### CODE REVIEW

File: src/Keboola/DbExtractor/TableResultFormat/ForeignKey.php [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+    {
+        $this->refColumn = $refColumn;
+        return $this;
+    }
+
+    public function getOutput(): array
+    {
+        $ret = [];
+        foreach (get_class_vars(self::class) as $property => $propertyValue) {
+            $propertyValue = $this->{$property} ?? null;
```
Comment (pivnicek) [inline src/Keboola/DbExtractor/TableResultFormat/ForeignKey.php]: je to potreba?  kdy `$this->property` nema value je to `null` automaticke ne?

File: src/Keboola/DbExtractor/TableResultFormat/Table.php [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+    {
+        $this->columns[] = $column;
+        return $this;
+    }
+
+    public function getOutput(): array
+    {
+        $ret = [];
+        foreach (get_class_vars(self::class) as $property => $propertyValue) {
+            $propertyValue = $this->{$property} ?? null;
```
Comment (pivnicek) [inline src/Keboola/DbExtractor/TableResultFormat/Table.php]: jako nahoru

File: src/Keboola/DbExtractor/TableResultFormat/Table.php [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+            if ($property === 'requiredParams') {
+                continue;
+            }
```
Comment (pivnicek) [inline src/Keboola/DbExtractor/TableResultFormat/Table.php:68]: asi nejlepsi tohle delat jako prvni vec.  Kdy `$property === 'requiredParams'` za mne $propertyValue nema smysl

File: src/Keboola/DbExtractor/TableResultFormat/TableColumn.php [hunk 4]
```diff
@@ -0,0 +1,128 @@
...
+
+    private bool $nullable;
+
+    private string $default;
+
+    private int $ordinalPosition;
+
+    private string $description;
+
+    private bool $autoIncrement;
```
Comment (pivnicek) [inline src/Keboola/DbExtractor/TableResultFormat/TableColumn.php:32]: myslim se ze tohle uz nepouzivame?

File: tests/Keboola/TableResultFormatTest.php [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+    public function testInvalidTable(): void
+    {
+        $table = new Table();
+        $table
+            ->setCatalog('catalog');
+
+        $this->expectException(UserException::class);
+        $this->expectExceptionMessage('Parameter \'name\' is required');
+        $table->getOutput();
+    }
```
Comment (pivnicek) [inline tests/Keboola/TableResultFormatTest.php:109]: Missing test for ForeignKey object

**Review by pivnicek — CHANGES_REQUESTED** [review]
Cool!  Bude to daleko lepsi :-)

Par veci a asi + 1 test.

File: src/Keboola/DbExtractor/TableResultFormat/ForeignKey.php (hunk 1 above)
Comment (ondrajodas) [reply src/Keboola/DbExtractor/TableResultFormat/ForeignKey.php]: v php 7.4 když jsem zkusil `$this->property` tak jsem dostal chybu
image

File: src/Keboola/DbExtractor/TableResultFormat/TableColumn.php (hunk 4 above)
Comment (ondrajodas) [reply src/Keboola/DbExtractor/TableResultFormat/TableColumn.php:32]: používá se např:
https://github.com/keboola/db-extractor-mssql/blob/master/src/Keboola/DbExtractor/Extractor/MetadataProvider.php#L264

ale i v jiných extractorech

File: src/Keboola/DbExtractor/TableResultFormat/Table.php (hunk 3 above)
Comment (ondrajodas) [reply src/Keboola/DbExtractor/TableResultFormat/Table.php]: added

File: tests/Keboola/TableResultFormatTest.php (hunk 5 above)
Comment (ondrajodas) [reply tests/Keboola/TableResultFormatTest.php:109]: added

File: src/Keboola/DbExtractor/TableResultFormat/TableColumn.php (hunk 4 above)
Comment (pivnicek) [reply src/Keboola/DbExtractor/TableResultFormat/TableColumn.php:32]: jo, but what I meant was that I initial used this as part of incremental fetching logic but it is not used that way anymore.  Maybe it is still interesting metadata though so we can keep it.

**Review by pivnicek — APPROVED** [review]
parada

**Pull request merged**